### DISCUSSION TITLE
Any example how to stop the streamming

### DISCUSSION THREAD

**Discussion by jayhuang75** [tesaguri/twitter-stream-rs#30] [Q&A]
We are looking for an example function to stop the streaming, any guidance or suggestion?

**Comment by tesaguri** [comment]
Sorry for the delay. The issue seems to have somehow slipped out of e-mail notifications.

---

Like other `impl Stream` types in general, just dropping the `TwitterStream` value should cancel the task. The possible way of dropping it depends on how you yield its values. Here is a pair of examples:

- If you are running a `while let Some(str) = stream.next()` loop, you can just drop the stream after breaking the loop
- If you are using a `{Try,}StreamExt` combinator to produce an `impl Future` value and `await`-ing for it like `stream.try_for_each(/* ... */).await`, you may need to use another combinator like [`select()`] to interrupt the task

[`select()`]: <https://docs.rs/futures-util/0.3.21/futures_util/future/fn.select.html>
